service/src/main/java/bio/terra/drshub/services/DrsResolutionService.java

```diff
@@ -44,17 +46,21 @@ public class DrsResolutionService {
   private final DrsProviderService drsProviderService;
   private final AuthService authService;
   private final AuditLogger auditLogger;
+  private final UserLoggingMetrics userLoggingMetrics;
+  private static final String TRANSACTION_ID_FIELD_NAME = "transactionId";
 
   @Autowired
   public DrsResolutionService(
       DrsApiFactory drsApiFactory,
       DrsProviderService drsProviderService,
       AuthService authService,
-      AuditLogger auditLogger) {
+      AuditLogger auditLogger,
+      UserLoggingMetrics userLoggingMetrics) {
     this.drsApiFactory = drsApiFactory;
     this.drsProviderService = drsProviderService;
     this.authService = authService;
     this.auditLogger = auditLogger;
+    this.userLoggingMetrics = userLoggingMetrics;
```

Comment:
DRSHub already logs its DRS Resolutions via the Audit Logger. I think we should include the transaction ID in that message, rather than publishing a separate message.

Reply:
This allows us to include the transaction id in the properties for existing Bard events (like how TDR does it). It's not creating an additional log message. I can include it in the audit log message as well though!

Reply:
Oh, do we need the `UserLoggingMetrics` so that we can inject more attributes into the Bard request?